R/FilterStates.R

```diff
@@ -273,7 +273,7 @@ FilterStates <- R6::R6Class( # nolint
         excluded_varnames <- setdiff(varnames, private$get_filterable_varnames())
         if (length(excluded_varnames)) {
           state <- Filter(function(x) !isTRUE(x$varname %in% excluded_varnames), state)
-          logger::log_warn("filters for columns: { toString(excluded_varnames) } excluded from { private$dataname }")
+          warning(sprintf("filters for columns: %s excluded from %s", toString(excluded_varnames), private$dataname))
```

Comment:
@pawelru you sure you want to substitute logger with regular warning? I though with logger you set a logging setup for the depth so that depending on the initial setup some warnings/errrors/logs will appear and some not. With a regular warning - this will always appear and there will be no option to turn this off if you dont want to see those becauase of some reasone.

Reply:
I have been thinking about this. Note that from the testing perspective when we put `expect_message(foo(), "WARNING.*blabla")` we are actually dependent on logger setting including layout, appender function, threshold and so on. We are actually testing if and how logger is passing raised warning other than whether warning had been raised. I would like to limit that dependency somehow hence this PR.
Please have a look https://github.com/insightsengineering/coredev-tasks/issues/502. I have found a promising functionality that could allow devs to use base syntax (simpler) while still having logger on (including threshold, custom appender, etc.).

Reply:
I think that my change should be in line with that task so let me assign myself and create a change request. Then we will be able to see it all in action.
Unfortunately I did this together with options issue - this should have been done separately so I could only now apologise for inconvenience and hope it's not a big problem for the reviewer.